sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -483,8 +483,8 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
    * 4. Otherwise, `this` and `that` are considered as conflicting schemas and an exception would be
    *    thrown.
    */
-  private[sql] def merge(that: StructType): StructType =
-    StructType.merge(this, that).asInstanceOf[StructType]
+  private[sql] def merge(that: StructType, resolver: Resolver = _ == _): StructType =
```

Comment:
Should we use `SQLConf.resolver` as default resolver?

Reply:
The default resolver looks like for case-sensitive resolution. SparkSQL is case-insensitive by default. I'm afraid calling it by default parameter could lead to unexpected error. I'm not sure why `findNestedField` uses the default resolver. Removing default resolve is also good to me.

Reply:
Done